src/FactSystem/ParameterLoader.cc

```diff
@@ -552,6 +583,106 @@ void ParameterLoader::_writeParameterRaw(int componentId, const QString& paramNa
     _vehicle->sendMessage(msg);
 }
 
+void ParameterLoader::_writeLocalParamCache()
+{
+    QMap<int, MapID2NamedParam> cache_map;
+
+    foreach(int component, _mapParameterId2Name.keys()) {
+        foreach(int id, _mapParameterId2Name[component].keys()) {
+            const QString name(_mapParameterId2Name[component][id]);
+            const Fact *fact = _mapParameterName2Variant[component][name].value<Fact*>();
+            cache_map[component][id] = NamedParam(name, ParamTypeVal(fact->type(), fact->value()));
+        }
+    }
+
+    QFile cache_file(QFileInfo(QSettings().fileName()).path() + QDir::separator() + "param_cache");
+    cache_file.open(QIODevice::WriteOnly | QIODevice::Truncate);
+
+    QDataStream ds(&cache_file);
+    ds << cache_map;
+}
+
+static MAV_PARAM_TYPE factType2MavParamType(int factType)
+{
+    MAV_PARAM_TYPE paramType;
+    switch (factType) {
+        case FactMetaData::valueTypeUint8:
+            paramType = MAV_PARAM_TYPE_UINT8;
+            break;
+        case FactMetaData::valueTypeInt8:
+            paramType =  MAV_PARAM_TYPE_INT8;
+            break;
+        case FactMetaData::valueTypeUint16:
+            paramType = MAV_PARAM_TYPE_UINT16;
+            break;
+        case FactMetaData::valueTypeInt16:
+            paramType = MAV_PARAM_TYPE_INT16;
+            break;
+        case FactMetaData::valueTypeUint32:
+            paramType = MAV_PARAM_TYPE_UINT32;
+            break;
+        case FactMetaData::valueTypeInt32:
+            paramType = MAV_PARAM_TYPE_INT32;
+            break;
+        case FactMetaData::valueTypeFloat:
+            paramType = MAV_PARAM_TYPE_REAL32;
+            break;
+        case FactMetaData::valueTypeDouble:
+            paramType = MAV_PARAM_TYPE_REAL64;
+            break;
+        default:
+            paramType = MAV_PARAM_TYPE_INT32;
+            qCritical() << "Unsupported fact type" << factType;
+            break;
+    }
+    return paramType;
+}
+
+void ParameterLoader::_tryCacheHashLoad(int uasId, QVariant hash_value)
+{
+    uint32_t crc32_value = 0;
+    /* The datastructure of the cache table */
+    QMap<int, MapID2NamedParam> cache_map;
+
+    QFile cache_file(QFileInfo(QSettings().fileName()).path() + QDir::separator() + "param_cache");
```

Comment:
I went with this based on what I found [here](https://github.com/mavlink/qgroundcontrol/blob/2ba8a0f3323bd91826873d58d4a6f610f6ae49e6/src/AutoPilotPlugins/PX4/PX4AirframeLoader.cc#L68) but I am open to modifying this.